src/ProgressOnderwijsUtils/SchemaReflection/DatabaseDescription.cs

```diff
@@ -137,13 +137,59 @@ internal Index(TObject containingObject, DbObjectIndex indexMetaData, DatabaseDe
             public Column<TObject> Column
                 => Index.ContainingObject.ColumnsById[UnderlyingMetaData.ColumnId];
 
+
+            public bool IsDescending
+                => UnderlyingMetaData.IsDescending;
         }
 
         public DbObjectId ObjectId
             => IndexMetaData.ObjectId;
 
         public DbIndexId IndexId
             => IndexMetaData.IndexId;
+
+        public SqlIndexType IndexType
+            => IndexMetaData.IndexType;
+
+        public string? IndexName
+            => IndexMetaData.IndexName;
+
+        public string IndexCreationScript()
+        {
+            var include = IndexType.IsColumnStore() || IncludedColumns.None() ? "" : $" include ({IncludedColumns.Select(col => col.Column.ColumnName).JoinStrings(", ")})";
+            var filter = Filter is null ? "" : $" where {SqlServerUtils.PrettifySqlExpression(Filter)}";
+            var compression = DataCompressionType is SqlCompressionType.None or SqlCompressionType.ColumnStore ? "" : $" with (data_compression={DataCompressionType.ToSqlName().CommandText()})";
+            var defaultIndexType = IsPrimaryKey ? SqlIndexType.ClusteredIndex : SqlIndexType.NonClusteredIndex;
+            var indexType = IndexType == defaultIndexType ? "" : " " + IndexType.ToSqlName().CommandText();
+            var indexColumns = IndexType == SqlIndexType.ClusteredColumnStore
+                ? ""
+                : $" ({IndexColumns.Select(col => col.Column.ColumnName + (col.IsDescending ? " desc" : "")).JoinStrings(", ")})";
+
+            if (IsPrimaryKey || IsUniqueConstraint) {
+                var constraintType = IsPrimaryKey ? " primary key" : " unique";
+                return $"alter table {ContainingObject.QualifiedName} add constraint {IndexName}{constraintType}{indexType}{indexColumns}{compression}";
+            } else if (IndexType == SqlIndexType.Heap) {
+                return $"-- {ContainingObject.QualifiedName} is a heap{compression}";
+            } else {
+                var unique = IsUnique ? " unique" : "";
+                return $"create{unique}{indexType} index {IndexName} on {ContainingObject.QualifiedName}{indexColumns}{include}{filter}{compression}";
```

Comment:
Waarom de spaties niet hierin en in de vars geen spaties?
Extra spaties maakt toch niet uit?